### PR TITLE
Windows Support

### PR DESCRIPTION
**Pull request by TomFreeman**
Slightly mega PR adding support for both windows, and debugging in vscode.

If necessary, I could unpick the debugging changes, or submit them separately.

Highlights of changes:

- [ ] addition of ts-node, and changes to `/bin/cli` to launch with ts-node if src repo is available (debugging support)
- [ ] addition of `.vscode/launch.json` file to specify debug configurations (debugging support)
- [ ] addition or `rimraf` and `cross-env` packages, and some tweaks to packages.json to make all scripts run x-plat (windows support)
- [ ] Addition of `upath` package, and replace `path` across the whole package with `upath`, and occasionally explicitly call `upath.toUnix` to convert file paths to unix paths (windows support)

Tested on windows 10, and ubuntu.

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -1,3 +1,30 @@
 #!/usr/bin/env node
 
-require('../build/src/cli.js')
+var devMode = require('fs').existsSync(`${__dirname}/../src`);
```
Comment (TomFreeman) [inline bin/cli.js:3]: This assumes that the src folder at the root isn't included in the built package.

I have in the past allowed a developer to specify they still want to use the compiled scripts when running locally, but it doesn't seem very useful.

File: src/Injector.ts [hunk 2]
```diff
@@ -16,8 +16,8 @@ class Injector {
    * Render the base template and inject content if provided
    */
   injectAndRender (inputPath: string, inputIndexYaml: string, boatsRc: BoatsRC): string {
-    const fullPath = path.join(process.cwd(), inputPath);
-    const relativePathToRoot = path.relative(path.dirname(inputPath), path.dirname(inputIndexYaml));
+    const fullPath = upath.join(upath.toUnix(process.cwd()), inputPath);
```
Comment (TomFreeman) [inline src/Injector.ts:19]: Not 100% sure it's required to wrap the `process.cwd` fall inside the toUnix call in all cases, but it's working at the moment like this.

Also, there's 4 occurances of this pattern `upath.toUnix(process.cwd())` - so probably should pull it out into a helper function.

File: tsconfig.json [hunk 3]
```diff
@@ -1,13 +1,17 @@
 {
+  "ts-node": {
+    "transpileOnly": true,
+    "compiler": "ttypescript"
+  },
   "compilerOptions": {
-    "target": "ES2018",
+    "target": "ES2019",
```
Comment (TomFreeman) [inline tsconfig.json:7]: These changes (`target` and `baseUrl`) are probably unnecessary, and a result of initial attempts to make debugging support work.

Happy to revert them.

File: bin/cli.js (hunk 1 above)
Comment (johndcarmichael) [reply bin/cli.js:3]: this is nice, but the npm ignore should then ignore the src folder

**Comment by johndcarmichael** [comment]
In general, it looks great and thanks again for the effort and commitment!

There was a PR that came in just before yours which is now tested and released, if your could merge in these changes as there appears to be a few conflicts.

The dev mode in the cli is cool, but unless i'm mistaken there needs to be an entry in the .npmignore for that to work.

Other than that, it looks good!

**Comment by TomFreeman** [comment]
Have resolved the merge conflicts and updated the npm ignore. Seems the built package still works.

File: .vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (johndcarmichael) [inline .vscode/launch.json:1]: I do not use vscode myself so I have no way of telling if this is good or bad :D

Comment (TomFreeman) [reply .vscode/launch.json:1]: image

Comment (johndcarmichael) [reply .vscode/launch.json:1]: :) well the comment in the JSON cannot be correct though.. syntactically incorrect...

Comment (TomFreeman) [reply .vscode/launch.json:1]: I can take them out, vscode itself put them there.

Comment (johndcarmichael) [reply .vscode/launch.json:1]: yea i think that would be cleaner - this is only replicating what is already in the package.json

Comment (TomFreeman) [reply .vscode/launch.json]: I've taken it out, wasn't sure about having editor-specific config in the repo to be honest.

**Comment by johndcarmichael** [comment]
All merged and release @TomFreeman - thank you for the effort, much appreciated!

**Pull request merged**